.github/workflows/build_msi_installer.yml

```diff
@@ -126,18 +128,27 @@ jobs:
         with:
           creds: ${{ secrets.AZURE_CREDENTIALS }}
 
+      - name: Download JSON file from Azure Blob Storage
+        id: download-json
+        run: |
+          az storage blob download --account-name promptflowartifact --container-name msi-installer --name latest_version.json --file downloaded_version.json
+          $downloaded_version=$(python -c "import json; new_version = json.load(open('downloaded_version.json'))['promptflow']")
+          echo "::set-output name=downloaded_version::downloaded_version"
+
       - name: Check if version input is valid and upload JSON file
         run: |
           $version = "${{ steps.get-version.outputs.version }}"
-          if ($version -like '1.*') {
+          $downloaded_version = "${{ steps.download-json.outputs.downloaded_version }}"
+          if ($version -like '1.*' -and $version -gt $downloaded_version){
```

Comment:
-gt cannot detect case like 
"1.10.1" -gt "1.9.0"

Reply:
fixed

Reply:
what's the fix here?

Reply:
![image](https://github.com/microsoft/promptflow/assets/26239730/29418c3c-e134-4959-a5da-45e875e29c8c)